src/source/Ice/Network.c

```diff
@@ -331,45 +331,129 @@ STATUS socketWrite(INT32 sockfd, const void* pBuffer, SIZE_T length)
     return retStatus;
 }
 
+BOOL isIpAddr(PCHAR hostname, UINT16 length)
+{
+    BOOL status = TRUE;
+    UINT32 ip_1, ip_2, ip_3, ip_4, ip_5, ip_6, ip_7, ip_8;
+    if (hostname == NULL || length > MAX_ICE_CONFIG_URI_LEN) {
+        DLOGW("Provided NULL hostname");
+        status = FALSE;
+    } else {
+        status = (SSCANF(hostname, IPV4_TEMPLATE, &ip_1, &ip_2, &ip_3, &ip_4) == 4 && ip_1 >= 0 && ip_1 <= 255 && ip_2 >= 0 && ip_2 <= 255 &&
+                  ip_3 >= 0 && ip_3 <= 255 && ip_4 >= 0 && ip_4 <= 255) ||
+            (SSCANF(hostname, IPV6_TEMPLATE, &ip_1, &ip_2, &ip_3, &ip_4, &ip_5, &ip_6, &ip_7, &ip_8) == 8);
+    }
+    return status;
+}
+
+STATUS getIpAddrFromDnsHostname(PCHAR hostname, PCHAR address, UINT16 lengthSrc, UINT16 maxLenDst)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT8 i = 0, j = 0;
+    UINT16 hostNameLen = lengthSrc;
+    CHK(hostname != NULL && address != NULL, STATUS_NULL_ARG);
+    CHK(hostNameLen > 0 && hostNameLen < MAX_ICE_CONFIG_URI_LEN, STATUS_INVALID_ARG);
+
+    // TURN server URLs conform with the public IPv4 DNS hostname format defined here:
```

Comment:
Should the logic to derive the IP from the hostname for turn-only DNS domains?

Reply:
It would pass only for turn domains. For STUN, it would fail the moment the first call is made. I can add a check to see if the server has stun in it, but did not bother because the parse would fail immediately.